docs/design.md

```diff
@@ -159,3 +152,92 @@ The workers are subscribed to this queue and process the data, then notify other
+When a new message is received, the broker composes a message according to the rasa server interface specifications and sends it to the Rasa server though an HTTP POST request. The response to the HTTP request contains the list of messages produced by the Rasa Bot. Upon the response reception, the broker sends each single message to the Niceday server using the sendTextMessage(id, text) function of the Chat object of [goalie-js](https://github.com/senseobservationsystems/goalie-js).
+
+<img src = "img/sequence.png" width = "500" title="Sequence diagram">
+
+### Data exchange with Niceday Server
+
+To retrieve data from the Niceday Server, the [Niceday API](https://github.com/PerfectFit-project/niceday-components/tree/main/niceday-api) has to be used. The  API is used by the Rasa actions, using the [NicedayClient](https://github.com/PerfectFit-project/niceday_client) Python interface. 
+
+<img src = "img/sequence_data.png" width = "500" title="Sequence diagram data">
+
+### Data exchange with local Database
+
+User’s data and output of the conversations can be stored to and retrieved from the local DB, represented here below. The interactions with the DB are performed by the rasa actions, using the [virtual-coach-db](https://github.com/PerfectFit-project/virtual-coach-db) package.
+
+<img src = "img/db.png" width = "300" title="Sequence diagram data">
+
+---
+
+## Interfaces specifications
+
+The different components exchange information one with the other using HTTP requests. The requests specifications are reported here below.
+N.B. The URLs  reported are assuming that the component is running using the Docker image generated using the [docker-compose file](https://github.com/PerfectFit-project/virtual-coach-main/blob/main/docker-compose.yml). If the components are running in isolation, or outside the Docker network, the URL has to be changed accordingly. 
+
+###Rasa server
+
+URL: http://rasa_server:5005/webhooks/rest/webhook
+
+Request type: POST
+
+Body:
+
+{
+    "sender":"sender_id",
+    "message":"text_message"
+}
+
+Output:
+
+[
+    {
+        "recipient_id": "id",
+        "text": "text_message"
+    },
+    {
+        "recipient_id": "id",
+        "text": " text_message”
+    }
+]
```

Comment:
Also please put this in a codeblock, so it is rendered nicely. See https://github.com/PerfectFit-project/virtual-coach-main/blob/documentation-revision/docs/design.md#interfaces-specifications